dotnet/src/Agents/OpenAI/OpenAIAssistantDefinition.cs

```diff
@@ -34,24 +34,49 @@ public sealed class OpenAIAssistantDefinition
     public string? Name { get; init; }
 
     /// <summary>
-    /// Set if code-interpreter is enabled.
+    /// Optional file-ids made available to the code_interpreter tool, if enabled.
     /// </summary>
-    public bool EnableCodeInterpreter { get; init; }
+    public IReadOnlyList<string>? CodeInterpreterFileIds { get; init; }
 
     /// <summary>
-    /// Set if retrieval is enabled.
+    /// Set if code-interpreter is enabled.
     /// </summary>
-    public bool EnableRetrieval { get; init; }
+    public bool EnableCodeInterpreter { get; init; }
 
     /// <summary>
-    /// A list of previously uploaded file IDs to attach to the assistant.
+    /// Set if json response-format is enabled.
     /// </summary>
-    public IEnumerable<string>? FileIds { get; init; }
+    public bool EnableJsonResponse { get; init; }
 
     /// <summary>
     /// A set of up to 16 key/value pairs that can be attached to an agent, used for
     /// storing additional information about that object in a structured format.Keys
     /// may be up to 64 characters in length and values may be up to 512 characters in length.
     /// </summary>
     public IReadOnlyDictionary<string, string>? Metadata { get; init; }
+
+    /// <summary>
+    /// The sampling temperature to use, between 0 and 2.
+    /// </summary>
+    public float? Temperature { get; init; }
```

Comment:
This class is a strong candidate for Serialization / Deserialization
```suggestion
    [JsonPropertyName("temperature")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public float? Temperature { get; init; }
```